src/pubsub-publisher/pubsub-publisher.service.ts

```diff
@@ -37,36 +37,36 @@ export class PubSubPublisherService {
     data: Buffer,
     attributes: Record<string, string>
   ): Promise<string> {
-    const timeout = setTimeoutAsync(this.settings.requestTimeoutMillis);
+    const publisherClient = new v1.PublisherClient({});
+    const formattedTopic = publisherClient.projectTopicPath(projectId, topic.name);
+    const dataBuffer = Buffer.from(data);
```

Comment:
Are you sure this is using the same pubsub config as the previous solution?

I think it creates a new PubSub instance in `new v1.PublisherClient({});`.

Also there is probably a way to get PublisherClient from `PubSub` or `Topic` instances.

Reply:
there might be a way to get publisher client from topic instance, was trying to apply that while fixing tests, will verify